AK/Bitmap.h

```diff
@@ -163,68 +158,141 @@ class Bitmap {
         return {};
     }
 
-    Optional<size_t> find_longest_range_of_unset_bits(size_t max_length, size_t& found_range_size) const
+    // The function will return the next range of unset bits starting from the
+    // @from value. 
+    // @from: the postition from which the search starts. The var will be
+    //        changed and new value is the offset of the found block.
+    // @min_length: minimum size of the range which will be returned.
+    // @max_length: maximum size of the range which will be returned.
+    //              This is used to increase performance, since the range of
+    //              unset bits can be long, and we don't need the while range,
+    //              so we can stop when we've reached @max_length.
+    inline Optional<size_t> find_next_range_of_unset_bits(size_t& from, size_t min_length=1, size_t max_length=max_size) const
     {
-        auto first_index = find_first_unset();
-        if (!first_index.has_value())
-            return {};
 
-        size_t free_region_start = first_index.value();
-        size_t free_region_size = 1;
+#ifndef count_trailing_zeroes
+        auto count_trailing_zeroes = [](u32 val) -> u32{
+            for (u8 i = 0; i < 32; ++i) {
+                if ((val >> i) & 1) {
+                    return i;
+                }
+            }
+            return 0;
+        };
+#endif // count_trailing_zeroes
```

Comment:
The point of moving this to Platform.h was to get rid of the inline definition here :)